src/ape/managers/query.py

```diff
@@ -101,14 +106,25 @@ def find_creation_block(lo, hi):
         if not self.provider.get_code(query.contract):
             return None
 
-        block = find_creation_block(0, self.chain_manager.blocks.height)
+        # requires archive node
+        try:
+            block = find_creation_block(0, self.chain_manager.blocks.height)
+        except ProviderError:
+            self.supports_contract_creation = False
+            return None
 
-        # iterate over transaction traces in a block to find the deploy transaction
+        # iterate over transaction traces in a block to find the deployment transaction
         # this method also supports contract factories
-        if "geth" in self.provider.client_version.lower():
-            yield from self._find_creation_in_block_via_geth(block, query.contract)
-        else:
-            yield from self._find_creation_in_block_via_parity(block, query.contract)
+        try:
+            if "geth" in self.provider.client_version.lower():
+                yield from self._find_creation_in_block_via_geth(block, query.contract)
```

Comment:
Some of this seems like it'd make more sense in `ape-ethereum`, since it uses EVM-based RPCs.

Reply:
should i join all methods and move them under `ape_geth.query`?

Reply:
did a bit of a refactor:
- move the binary search query from the default engine into ethereum engine
- move otterscan query from geth to otterscan engine